tests/unit/_modules/test_cephprocesses.py

```diff
@@ -0,0 +1,506 @@
+        proc = self.build_proc(role_name, proc_name)
+        mc.up = [proc]
+        mc.down = [proc_name_down]
+        assert mc.report() == {'up': {'ceph-osd': [0]}, 'down': {'ceph-mds': 'ceph-mds', 'ceph-osd': ['1', '3']}}
+
+
+class TestInstanceMeth():
+
+    def test_check(self):
+        """
+        Exit if there is no pillar data
+        """
+        cephprocesses.__pillar__ = {}
+        assert cephprocesses.check() == False
+
+    @mock.patch('srv.salt._modules.cephprocesses.psutil')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck')
+    @mock.patch('srv.salt._modules.cephprocesses.ProcInfo')
+    def test_check_1(self, proc_mock, meta_mock, psutil_mock):
+        """
+        Exit if there is pillar data but no roles in it
+        """
+        cephprocesses.__pillar__ = {'NOroles': ['dummy']}
+        assert cephprocesses.check() == False
+
+    @pytest.mark.parametrize("test_input,expected", [
+    ("mgr", 'mgr'),
+    ("mon", 'mon'),
+    ("mds", 'mds'),
+    ("openattic", 'openattic'),
+    ("rgw", 'rgw'),
+    ("benchmark-rbd", 'benchmark-rbd'),
+    ("ganesha", 'ganesha')
+    ])
+    @mock.patch('srv.salt._modules.cephprocesses.psutil.process_iter')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.report')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.check_absents')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.check_inverts')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.check_osds')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.add')
+    @mock.patch('srv.salt._modules.cephprocesses.ProcInfo')
+    def test_check_2(self, proc_mock, meta_mock, mock_check_osds, meta_check_invert, meta_check_absent, mock_report, psutil_mock, test_input, expected):
+        """
+        Parameterized test to verify that roles get passed down to methods
+        """
+        role = [test_input]
+        cephprocesses.__pillar__ = {'roles': role}
+        psutil_mock.return_value = ['proc1']
+        cephprocesses.check()
+        proc_mock.assert_called_with('proc1')
+        # ANY instance of ProcInfo
+        meta_mock.assert_called_with(ANY, expected)
+        meta_check_invert.assert_called_with(expected)
+        meta_check_absent.assert_called_with(expected)
+        mock_check_osds.assert_called_once is False
+        mock_report.assert_called
+
+    @mock.patch('srv.salt._modules.cephprocesses.psutil.process_iter')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.report')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.check_absents')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.check_inverts')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.check_osds')
+    @mock.patch('srv.salt._modules.cephprocesses.MetaCheck.add')
+    @mock.patch('srv.salt._modules.cephprocesses.ProcInfo')
+    def test_check_3(self, proc_mock, meta_mock, mock_check_osds, meta_check_invert, meta_check_absent, mock_report, psutil_mock):
```

Comment:
Does it make sense to add 'storage' role to the test_check_2 and only wrap check_osds.assert_called_once into if ... else check?
Separate method is a bit of an overkill.

Reply:
let me try that

Reply:
Seems good, unless I forgot something. I'll go with that